pluggy/__init__.py

```diff
@@ -664,7 +664,11 @@ def _add_hookimpl(self, hookimpl):
     def __repr__(self):
         return "<_HookCaller %r>" % (self.name,)
 
-    def __call__(self, **kwargs):
+    def __call__(self, *args, **kwargs):
+        if args:
+            raise TypeError(
+                "hookimpl takes keyword arguments only".format(self.name)
```

Comment:
Haha yeah.. I changed it 3 times so I'm not surprised.
Thanks for catching / I'll fix it.